### PR TITLE
[5.0] Rename Login with Web Authentication to Login with Passkeys

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #41077 .

### Summary of Changes
Change verbage and change image used in the admin 
- note for b/c this adds a new image instead of replacing an existing one
- this is made against 5.0 as technicaly its a breaking change

### Testing Instructions
Code review the text to make sure that it still _makes sense_
- note that its tricky to know when it should be passkey and when passkeys - I went with my gut

cc @nikosdion 

All docs will need updating/reviewing if this PR is accepted

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [x] Pull Request link for manual.joomla.org: <[link](https://github.com/joomla/Manual/pull/142)>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/language/en-GB/plg_system_webauthn.ini [hunk 1]
```diff
@@ -3,49 +3,49 @@
 ; License GNU General Public License version 2 or later; see LICENSE.txt
 ; Note : All ini files need to be saved as UTF-8
 
-PLG_SYSTEM_WEBAUTHN="System - WebAuthn Passwordless Login"
-PLG_SYSTEM_WEBAUTHN_CANNOT_ADD_FOR_A_USER="You cannot add or remove authenticators on behalf of users. Users must login, and set up their own devices."
-PLG_SYSTEM_WEBAUTHN_DESCRIPTION="Enables passwordless authentication using the W3C Web Authentication (WebAuthn) API. Please note that the WebAuthn tab in the user profile editor and the WebAuthn login buttons will only be displayed if the user is accessing the site over HTTPS. Furthermore, registering WebAuthn authenticators and using them to log into your site will only work when your site is using a valid certificate, signed by a Certificate Authority the user's browser trusts."
+PLG_SYSTEM_WEBAUTHN="System - Passkeys Passwordless Login"
```
Comment (nikosdion) [inline administrator/language/en-GB/plg_system_webauthn.ini]: This should be singular. Maybe for better clarity have it as "System - Passkey (Passwordless) Login)".

File: administrator/language/en-GB/plg_system_webauthn.ini [hunk 2]
```diff
@@ -3,49 +3,49 @@
...
+PLG_SYSTEM_WEBAUTHN_CANNOT_ADD_FOR_A_USER="You cannot add or remove Passkeys on behalf of users. Users must login, and set up their own devices."
+PLG_SYSTEM_WEBAUTHN_DESCRIPTION="Enables passwordless authentication using browser Passkeys. Please note that the Passkeys tab in the user profile editor and the Passkeys login buttons will only be displayed if the user is accessing the site over HTTPS. Registering Passkeys and using them to log into your site will only work when your site is using a valid certificate, signed by a Certificate Authority the user's browser trusts."
```
Comment (nikosdion) [inline administrator/language/en-GB/plg_system_webauthn.ini]: `Enables passwordless authentication using browser Passkeys` change to `Enables passwordless authentication using Passkeys`. The "browser" part is implied by definition.

File: administrator/language/en-GB/plg_system_webauthn.ini [hunk 3]
```diff
@@ -3,49 +3,49 @@
...
+PLG_SYSTEM_WEBAUTHN_ERR_CREATE_INVALID_USER="For security reasons you are not allowed to register Passkeys on behalf of another user."
+PLG_SYSTEM_WEBAUTHN_ERR_CREATE_NO_ATTESTED_DATA="Something went wrong but no further information about the error is available at this time. Please retry registering your Passkey."
+PLG_SYSTEM_WEBAUTHN_ERR_CREATE_NO_PK="The server has not issued a Public Key for Passkey registration but somehow received a Passkey registration request from the browser. This means that someone tried to hack you or something is broken."
 PLG_SYSTEM_WEBAUTHN_ERR_CREDENTIAL_ID_ALREADY_IN_USE="Cannot save credentials. These credentials are already being used by a different user."
-PLG_SYSTEM_WEBAUTHN_ERR_EMPTY_USERNAME="You need to enter your username (but NOT your password) before selecting the Web Authentication login button."
+PLG_SYSTEM_WEBAUTHN_ERR_EMPTY_USERNAME="You need to enter your username (but NOT your password) before selecting the Passkeys login button."
```
Comment (nikosdion) [inline administrator/language/en-GB/plg_system_webauthn.ini]: Singular, not plural.

File: administrator/language/en-GB/plg_system_webauthn.ini [hunk 4]
```diff
@@ -3,49 +3,49 @@
...
-PLG_SYSTEM_WEBAUTHN_FIELD_LABEL="W3C Web Authentication (WebAuthn) Login"
-PLG_SYSTEM_WEBAUTHN_FIELD_N_AUTHENTICATORS_REGISTERED="%d WebAuthn authenticators already set up: %s"
-PLG_SYSTEM_WEBAUTHN_FIELD_N_AUTHENTICATORS_REGISTERED_0="No WebAuthn authenticator has been set up yet"
-PLG_SYSTEM_WEBAUTHN_FIELD_N_AUTHENTICATORS_REGISTERED_1="One WebAuthn authenticator already set up: %2$s"
-PLG_SYSTEM_WEBAUTHN_HEADER="W3C Web Authentication (WebAuthn) Login"
-PLG_SYSTEM_WEBAUTHN_LBL_DEFAULT_AUTHENTICATOR="Generic Authenticator"
+PLG_SYSTEM_WEBAUTHN_FIELD_DESC="Lets you manage passwordless login methods using browser Passkeys. You need a supported browser and Passkey (eg Google Chrome or Firefox with a FIDO2 certified security key).<br><br><strong>MacOS/iOS/watchOS:</strong> Touch/Face ID.<br><strong>Windows:</strong> Hello (Fingerprint / Facial Recognition / PIN).<br><strong>Android:</strong> Biometric screen lock.<br><br>You can find more details in the <a href=\"https://docs.joomla.org/Special:MyLanguage/WebAuthn_Passwordless_Login\" target=\"_blank\" rel=\"noopener noreferrer\">Passkey Passwordless Login documentation</a>."
```
Comment (nikosdion) [inline administrator/language/en-GB/plg_system_webauthn.ini]: Remove "browser", as per earlier comment.

File: administrator/language/en-GB/plg_system_webauthn.sys.ini [hunk 5]
```diff
@@ -3,5 +3,5 @@
 ; License GNU General Public License version 2 or later; see LICENSE.txt
 ; Note : All ini files need to be saved as UTF-8
 
-PLG_SYSTEM_WEBAUTHN="System - WebAuthn Passwordless Login"
-PLG_SYSTEM_WEBAUTHN_DESCRIPTION="Enables passwordless authentication using the W3C Web Authentication (WebAuthn) API. Please note that the WebAuthn tab in the user profile editor and the WebAuthn login buttons will only be displayed if the user is accessing the site over HTTPS. Furthermore, registering WebAuthn authenticators and using them to log into your site will only work when your site is using a valid certificate, signed by a Certificate Authority the user's browser trusts."
+PLG_SYSTEM_WEBAUTHN="System - Passkeys Passwordless Login"
```
Comment (nikosdion) [inline administrator/language/en-GB/plg_system_webauthn.sys.ini]: Singular, not plural, as per earlier comment.

File: administrator/language/en-GB/plg_system_webauthn.sys.ini [hunk 6]
```diff
@@ -3,5 +3,5 @@
...
+PLG_SYSTEM_WEBAUTHN_DESCRIPTION="Enables passwordless authentication using browser Passkeys. Please note that the Passkeys tab in the user profile editor and the Passkeys login buttons will only be displayed if the user is accessing the site over HTTPS. Registering Passkeys and using them to log into your site will only work when your site is using a valid certificate, signed by a Certificate Authority the user's browser trusts."
```
Comment (nikosdion) [inline administrator/language/en-GB/plg_system_webauthn.sys.ini]: Remove "browser" as per earlier comment

File: build/media_source/plg_multifactorauth_webauthn/images/passkey.svg
Comment (nikosdion) [inline build/media_source/plg_multifactorauth_webauthn/images/passkey.svg:1]: This is the wrong logo. The Passkey logo is https://media.fidoalliance.org/wp-content/uploads/2022/09/Passkey-Style-Guide.zip and you can download it from https://github.com/passkeydeveloper/passkeys.dev/blob/main/static/images/fido-passkey-black.svg (the links in the PDF are broken).

**Review by nikosdion — CHANGES_REQUESTED** [review]

**Comment by HLeithner** [comment]
Isn't Passkeys not only a implementation variant of Webauthn used as marketing term by MS, Apple, Google for Synced variant of webauthn ? Not sure if it make sense to rename it, maybe all add it as alternative name?

File: build/media_source/plg_multifactorauth_webauthn/images/passkey.svg
Comment (brianteeman) [reply build/media_source/plg_multifactorauth_webauthn/images/passkey.svg:1]: got the image - will work on it later

**Comment by brianteeman** [comment]
> Isn't Passkeys not only a implementation variant of Webauthn used as marketing term by MS, Apple, Google for Synced variant of webauthn ? Not sure if it make sense to rename it, maybe all add it as alternative name?

See original issue for explanation

**Comment by nikosdion** [comment]
@HLeithner It is, but this is really how people will be expected to use it. In a future Joomla version (maybe 6?) we can modify the passwordless login plugin to request only resident keys from the browser. This will remove the requirement for entering the username.

The reason we had not implemented this yet is that hardware authenticators have a very limited memory for resident keys (Yubico only adds 10 slots in their keys) and back in early 2020 there wasn't even a way to manage them. With the introduction of passkeys this limitation is lifted since the TPM-backed software authenticator in modern browsers and OS can store an unlimited number of keys — and even sync them across devices (and, very soon, across _device ecosystems_). This will make hardware authenticators largely obsolete for the use cases we have for Joomla.

At the same time, nobody prevents someone from creating an alternative login or MFA plugin which works with generic WebAuthn and hardware authenticators — or, better yet, allow pinning login to _specific_ makes and models of hardware authenticators, something which might be  necessary for large organisations with strict regulatory controls.

So, everyone wins in the long run.

**Comment by HLeithner** [comment]
Based on the style guild "passkey" should be written lowercase (in our case we have some position where we write nouns uppercase even if it's not at the beginning of a sentence), the "passkeys" variant seems to only used if you reference multiple keys. Can you please change to this following the style guide?

**Comment by brianteeman** [comment]
The style guide says 

> The word “passkey” is a common noun (think of it the way
you would refer to “password”), and should be written in
lowercase except when beginning a sentence

I have updated the pull request to reflect that. Thanks for pointing it out. Also added it to the style guide https://github.com/joomla/Manual/pull/142

**Review by tecpromotion — APPROVED** [review]

**Comment by tecpromotion** [comment]
I have tested this item :white_check_mark: successfully on 7b4e0c1c1ed7c4e7228c7269c14db7041d66bc97<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41085">issues.joomla.org/tracker/joomla-cms/41085</a>.</sub>

**Comment by HLeithner** [comment]
I will make 2 changes now and then I would merge it if you are ok with them

**Comment by HLeithner** [comment]
Hmm created a pr https://github.com/brianteeman/joomla-cms/pull/512 but still this image is missing 
image

**Comment by brianteeman** [comment]
> Hmm created a pr [brianteeman#512](https://github.com/brianteeman/joomla-cms/pull/512) but still this image is missing !

did you run the build scripts to make sure the image gets moved to the correct folder?

**Comment by brianteeman** [comment]
you forgot to update plugins\multifactorauth\webauthn\src\Extension\Webauthn.php

```
98,88:                    
 'image'              => 'media/plg_multifactorauth_webauthn/images/webauthn.svg',
```

**Comment by HLeithner** [comment]
> you forgot to update plugins\multifactorauth\webauthn\src\Extension\Webauthn.php
> 
> ```
> 98,88:                    
>  'image'              => 'media/plg_multifactorauth_webauthn/images/webauthn.svg',
> ```

create new images (which is needed for this) image is not my favorite task ;-)

**Comment by nikosdion** [comment]
@HLeithner Here's the official Passkey logo as an SVG:

```xml
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" width="1.5em"><circle cx="10.5" cy="6" r="4.5"/><path d="m22.5 10.5a3.5 3.5 0 1 0 -5 3.15v5.35l1.5 1.5 2.5-2.5-1.5-1.5 1.5-1.5-1.24-1.24a3.5 3.5 0 0 0 2.24-3.26zm-3.5 0a1 1 0 1 1 1-1 1 1 0 0 1 -1 1z"/><path d="m14.44 12.52a6 6 0 0 0 -2.44-.52h-3a6 6 0 0 0 -6 6v2h13v-5.51a5.16 5.16 0 0 1 -1.56-1.97z"/></svg>
```

**Comment by HLeithner** [comment]
@brianteeman sorry I think pushed to the wrong branch but the images is now included thanks to @crystalenka

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**